pkg/analyzer/analyzer.go

```diff
@@ -2,136 +2,157 @@ package analyzer
+	"github.com/k8sgpt-ai/k8sgpt/pkg/analyzer/pod"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/analyzer/pvc"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/analyzer/rs"
+	"github.com/k8sgpt-ai/k8sgpt/pkg/analyzer/service"
 	"github.com/k8sgpt-ai/k8sgpt/pkg/kubernetes"
-	"github.com/spf13/viper"
 )
 
-var coreAnalyzerMap = map[string]func(ctx context.Context, config *AnalysisConfiguration,
-	client *kubernetes.Client, aiClient ai.IAI, analysisResults *[]Analysis) error{
-	"Pod":                   AnalyzePod,
-	"ReplicaSet":            AnalyzeReplicaSet,
-	"PersistentVolumeClaim": AnalyzePersistentVolumeClaim,
-	"Service":               AnalyzeEndpoints,
-	"Ingress":               AnalyzeIngress,
-}
-
-var additionalAnalyzerMap = map[string]func(ctx context.Context, config *AnalysisConfiguration,
-	client *kubernetes.Client, aiClient ai.IAI, analysisResults *[]Analysis) error{
-	"HorizontalPodAutoScaler": AnalyzeHpa,
+type IAnalyzer interface {
+	Analyze() error
+	GetResult() []common.Result
 }
 
-func RunAnalysis(ctx context.Context, filters []string, config *AnalysisConfiguration,
-	client *kubernetes.Client,
-	aiClient ai.IAI, analysisResults *[]Analysis) error {
-
-	activeFilters := viper.GetStringSlice("active_filters")
+const (
+	PodAnalyzerName                   = "Pod"
+	ReplicaSetAnalyzerName            = "ReplicaSet"
+	PersistentVolumeClaimAnalyzerName = "PersistentVolumeClaim"
+	ServiceAnalyzerName               = "Service"
+	IngressAnalyzerName               = "Ingress"
+	HPAAnalyzerName                   = "HorizontalPodAutoScaler"
+)
 
-	analyzerMap := getAnalyzerMap()
+var (
+	coreAnalyzerList = []string{
+		PodAnalyzerName,
+		ReplicaSetAnalyzerName,
+		PersistentVolumeClaimAnalyzerName,
+		ServiceAnalyzerName,
+		IngressAnalyzerName,
+		HPAAnalyzerName,
+	}
 
-	// if there are no filters selected and no active_filters then run all of them
-	if len(filters) == 0 && len(activeFilters) == 0 {
-		for _, analyzer := range analyzerMap {
-			if err := analyzer(ctx, config, client, aiClient, analysisResults); err != nil {
-				return err
-			}
-		}
-		return nil
+	additionalAnalyzers = []string{
+		HPAAnalyzerName,
 	}
+)
 
-	// if the filters flag is specified
-	if len(filters) != 0 {
-		for _, filter := range filters {
-			if analyzer, ok := analyzerMap[filter]; ok {
-				if err := analyzer(ctx, config, client, aiClient, analysisResults); err != nil {
-					return err
-				}
-			}
-		}
-		return nil
+func NewAnalyzer(analyzer string, client *kubernetes.Client, context context.Context, namespace string) (IAnalyzer, error) {
+	analyzerConfig := common.Analyzer{
+		Namespace: namespace,
+		Context:   context,
+		Client:    client,
 	}
 
-	// use active_filters
-	for _, filter := range activeFilters {
-		if analyzer, ok := analyzerMap[filter]; ok {
-			if err := analyzer(ctx, config, client, aiClient, analysisResults); err != nil {
-				return err
-			}
-		}
+	analyzerConfig.PreAnalysis = make(map[string]common.PreAnalysis)
+
+	switch analyzer {
```

Comment:
I personally would have added a `GetName()` on the interface to avoid a switch case here